applications/visualizer/backend/tests/ingestion/test_validator.py

```diff
@@ -0,0 +1,463 @@
+    with pytest.raises(ValidationError):
+        Annotation.model_validate(data)
+
+
+instanciate_valid_data_tc: List[Callable[..., Data]] = [
+    lambda: Data(
+        neurons=[
+            Neuron(
+                inhead=True,
+                name="ADAL",
+                emb=True,
+                nt="l",
+                intail=False,
+                classes="ADA",
+                typ="i",
+            ),
+            Neuron(
+                inhead=True,
+                name="ADAR",
+                emb=True,
+                nt="l",
+                intail=False,
+                classes="ADA",
+                typ="i",
+            ),
+        ],
+        datasets=[
+            Dataset(
+                id="white_1986_jse",
+                name="White et al., 1986, JSE (adult)",
+                type=DatasetType.TAIL,
+                time=60,
+                visualTime=50,
+                description="Adult legacy tail with pre-anal ganglion",
+            )
+        ],
+        connections={
+            "white_1986_jse": [
+                Connection(
+                    ids=[6949667],
+                    post="ADAR",
+                    post_tid=[6693872],
+                    pre="ADAL",
+                    pre_tid=[6719648],
+                    syn=[1],
+                    typ=ConnectionType.CHEMICAL,
+                )
+            ]
+        },
+        annotations={
+            "head": Annotation(
+                root={
+                    "increase": [
+                        ("ADAL", "RIPL"),
+                        ("ADAR", "RIPR"),
+                        ("ADEL", "AVKR"),
+                    ],
+                    "postembryonic": [
+                        ("ADAL", "RMFL"),
+                    ],
+                }
+            ),
+            "complete": Annotation(),
+        },
+    ),
+]
+
+
+@pytest.mark.parametrize("data_fn", instanciate_valid_data_tc)
+def test__valid_data(data_fn: Callable[..., Data]):
+    # every data definition should be able to correctly
+    # be instanciated
+    data_fn()
+
+
+instanciate_invalid_data_tc: List[Callable[..., Data]] = [
+    lambda: Data(
+        neurons=[],
+        datasets=[],
+        connections={"white_1986_jse": []},  # dataset id not in datasets
+    ),
+    lambda: Data(
+        neurons=[],
+        datasets=[],
+        connections={},
+        annotations={
+            # headd is not a valid annotation key
+            "headd": Annotation(),  # type: ignore
+        },
+    ),
+]
+
+
+@pytest.mark.parametrize("data_fn", instanciate_invalid_data_tc)
+def test__invalid_data(data_fn: Callable[..., Data]):
+    # every data definition should raise an exception upon
+    # being instanciated
+    with pytest.raises(ValidationError):
+        data_fn()
+
+
+@pytest.fixture
+def data_fixture(request: pytest.FixtureRequest) -> JSON:
+    # TODO: can we move fixtures close to the test? i.g Path(request.fspath).parent / Path("fixtures")
```

Comment:
I'm not sure it's worth validating the whole files that are present in `data/db-raw-data`, I would more create a folder in the `tests/ingestion` folder with a subset of the files and values by file from `data/db-raw-data`.

Reply:
Yup, I agree that is a bit overkill to validate the whole data, but it is also so easy and nice to test agains real reference data xD

Reply:
But I'm ok with this! I will extract a subset of the data. I assume that I need to be careful to only specify neurons defined in `neurons.json`, right? It is not a rule in the ingestion format document for I assume that this is the case.